src/run.ts

```diff
@@ -47,10 +52,156 @@ const createRelease = async (
   }
 };
 
+const getSearchQueries = (base: string, commits: string[]) => {
+  return commits.reduce((searches, commit) => {
+    const lastSearch = searches[searches.length - 1];
+
+    if (lastSearch && lastSearch.length + commit.length <= 256 - 6) {
+      searches[searches.length - 1] = `${lastSearch}+hash:${commit}`;
+    } else {
+      searches.push(`${base}+hash:${commit}`);
+    }
+
+    return searches;
+  }, [] as string[]);
+};
+
+/* Comment on released Pull Requests/Issues  */
+const createReleaseComments = async(
+  octokit: ReturnType<typeof github.getOctokit>,
+  { tagName, htmlUrl }: { tagName: string; htmlUrl: string }
+) => {
+      /*
+      Here are the following steps to retrieve the released PRs and issues.
+    
+        1. Retrieve the tag associated with the release
+        2. Take the commit sha associated with the tag
+        3. Retrieve all the commits starting from the tag commit sha
+        4. Retrieve the PRs with commits sha matching the release commits
+        5. Map through the list of commits and the list of PRs to
+           find commit message or PRs body that closes an issue and
+           get the issue number.
+        6. Create a comment for each issue and PR
+      */
```

Comment:
Each released thing should be associated with a changeset. I know that it's still possible to just release something while piggybacking on the already prepared release and close something that way but... should we care about it? Shouldn't we guide users to the best practices?

What if we'd make this changeset-based and just appropriately mix that info with additional data (like the link to the release)? We could also add contents of the relevant changesets to the generated comment.

Reply:
I agree with that!

So the idea would be to do that in the `runVersion()` method when we have changesets.   
In the `readChangesetState()` response we can retrieve the changeset releases:

```ts
// changesets: { id, summary, releases }[]
let { preState, changesets } = await readChangesetState(cwd);
```

We can then base our logic on these changesets releases.